File: NCore/Commands/Commandbroadcast.cs
using NCore.netcraft.server.api;
using System.Threading.Tasks;

namespace NCore
{
    public class Commandbroadcast : Command
    {
        public Commandbroadcast() : base("broadcast", NCore.GetNCore().lang.get("commands.broadcast.description"), NCore.GetNCore().lang.get("commands.broadcast.usage"))
        {
        }

        public override async Task<bool> OnCommand(CommandSender sender, Command cmd, string[] args, string label)
        {
            NCore.Lang lang = sender.IsPlayer ? ((NetcraftPlayer)sender).lang : NCore.GetNCore().lang;
            if (!sender.GetAdmin())
            {
                await sender.SendMessage(lang.get("commands.generic.error.no-perms"));
                return true;
            }

            if (args.Length > 0)
            {
                await Netcraft.Broadcast($"[{sender.GetName()}] " + string.Join(" ", args));
                return true;
            }

            return false;
        }
    }
}

File: NetcraftServer/Commands/Commandbroadcast.vb
Imports NetcraftServer

Public Class Commandbroadcast
    Inherits Command
    Sub New()
        MyBase.New("broadcast", "Отправляет сообщение в чат", "broadcast <сообщение>")
    End Sub
    Public Overrides Function OnCommand(sender As CommandSender, cmd As Command, args() As String, label As String) As Boolean
        If Not sender.GetAdmin Then
            sender.SendMessage("У Вас недостаточно прав!")
            Return True
        End If
        If args.Length > 0 Then
            netcraft.server.api.NCSApi.Netcraft.Broadcast("[Broadcast] " + String.Join(" ", args))
            Return True
        End If
        Return False
    End Function
End Class
